### DISCUSSION TITLE
my foundry unable to recognize any command containing "-m"

### DISCUSSION THREAD

**Discussion by Dorane-Star** [Cyfrin/foundry-full-course-cu#3139] [Q&A]
for example, i several several commands containing "-m", the last one being :
```
forge snapshot -m testWithdrawFromMultipleFunders
```
everytime i get this error:

```
error: unexpected argument '-m' found

  tip: to pass '-m' as a value, use '-- -m'
```

@EngrPpis, plz help

**Comment by EngrPips** [comment]
Hello @Dorane-Star, use `--mt` instead, and let's see if the issue persists.

**Reply by Dorane-Star** [reply]
ok, thanks

**Reply by EngrPips** [reply]
Anytime.

**Reply by Dorane-Star** [reply]
The error diverted now, i hope the command works but i have some other errors somewhere

**Reply by Dorane-Star** [reply]
forge snapshot --mt testWithdrawFromMultipleFunders
Compiler run failed:
Error (7576): Undeclared identifier.
  --> src/PriceConverter.sol:24:37:
   |
24 |         uint256 ethPrice = getPrice(priceFeed);
   |                                     ^^^^^^^^^

Error: Compilation failed

**Reply by EngrPips** [reply]
Please show your code.

**Reply by Dorane-Star** [reply]
This is just too long, i don't know what to share and what not to

// SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;

import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol";

// Why is this a library and not abstract?
// Why not an interface?
library PriceConverter {
    // We could make this public, but then we'd have to deploy it
    function getPrice(
        AggregatorV3Interface priceFeed
    ) internal view returns (uint256) {
        // Sepolia ETH / USD Address
        // https://docs.chain.link/data-feeds/price-feeds/addresses
        (, int256 answer, , , ) = priceFeed.latestRoundData();
        // ETH/USD rate in 18 digit
        return uint256(answer * 10000000000);
    }

    // 1000000000
    function getConversionRate(
        uint256 ethAmount
    ) internal view returns (uint256) {
        uint256 ethPrice = getPrice(priceFeed);
        uint256 ethAmountInUsd = (ethPrice * ethAmount) / 1000000000000000000;
        // the actual ETH/USD conversion rate, after adjusting the extra 0s.
        return ethAmountInUsd;
    }
}

**Reply by EngrPips** [reply]
```js
/ SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;

import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol";

// Why is this a library and not abstract?
// Why not an interface?
library PriceConverter {
// We could make this public, but then we'd have to deploy it
function getPrice(
AggregatorV3Interface priceFeed
) internal view returns (uint256) {
// Sepolia ETH / USD Address
// https://docs.chain.link/data-feeds/price-feeds/addresses
(, int256 answer, , , ) = priceFeed.latestRoundData();
// ETH/USD rate in 18 digit
return uint256(answer * 10000000000);
}

// 1000000000
function getConversionRate(
    uint256 ethAmount
) internal view returns (uint256) {
    uint256 ethPrice = getPrice(priceFeed);
    uint256 ethAmountInUsd = (ethPrice * ethAmount) / 1000000000000000000;
    // the actual ETH/USD conversion rate, after adjusting the extra 0s.
    return ethAmountInUsd;
}
}
```

**Reply by EngrPips** [reply]
Where did you declare the `priceFeed` variable that you used in the line of code ` uint256 ethPrice = getPrice(priceFeed);` are you coding along with Patrick?

**Comment by AnaAse** [comment]
Substitute:   -m        with:  --match-test

**Reply by EngrPips** [reply]
Great heads-up
